dashboard/impl/src/java/org/sakaiproject/dash/logic/DashboardUserLogicImpl.java

```diff
@@ -108,19 +111,88 @@ public int countNewsLinksByGroupId(String sakaiUserId,
 							item.setTitle(typeObj.getGroupTitle(itemCount, item.getContext().getContextTitle(), item.getNewsTimeLabelKey()));
 						}
 					}
+				}else{
+					//When getItemCount() > 1 the infoLinkUrl is null and hence we don't need to run this call.
+					getInfoLinkUrlForNewsItem(item);
 				}
 			}
 		}
-		
+
 		return links;
 	}
 
+	private void getInfoLinkUrlForNewsItem(NewsItem item) {
```

Comment:
There are many common logic between getInfoLinkUrlForNewsItem and getInfoLinkUrlForCalenderItem. Please refactor the code.